src/ucp/core/ucp_ep.c

```diff
@@ -225,22 +227,52 @@ ucs_status_t ucp_ep_create_base(ucp_worker_h worker, const char *peer_name,
                       peer_name);
 #endif
 
+    if (!(ep_init_flags & UCP_EP_INIT_FLAG_INTERNAL) &&
+        ((context->config.ext.proto_indirect_id == UCS_CONFIG_ON) ||
+         ((context->config.ext.proto_indirect_id == UCS_CONFIG_AUTO) &&
+          (ep_init_flags & UCP_EP_INIT_ERR_MODE_PEER_FAILURE)))) {
+        ucp_ep_update_flags(ep, UCP_EP_FLAG_INDIRECT_ID, 0);
+    }
+
+    status = UCS_PTR_MAP_PUT(ep, &worker->ep_map, ep,
+                             !!(ep->flags & UCP_EP_FLAG_INDIRECT_ID),
```

Comment:
minor: no need for `!!`, API says it has to be zero or non-zero

Reply:
done